lux/core/frame.py

```diff
@@ -498,9 +498,15 @@ def maintain_recs(self, is_series="DataFrame"):
                 if len(vlist) > 0:
                     rec_df._recommendation[action_type] = vlist
             rec_df._rec_info = rec_infolist
+            quantitative_columns = [i for i in self.dtypes if i != "O" and i != "str"]
+            if len(quantitative_columns) == 2 or len(quantitative_columns) == 3:
+                self.current_vis = VisList([i for i in self.columns], self)
             self._widget = rec_df.render_widget()
         # re-render widget for the current dataframe if previous rec is not recomputed
         elif show_prev:
+            quantitative_columns = [i for i in self.dtypes if i != "O" and i != "str"]
+            if len(quantitative_columns) == 2 or len(quantitative_columns) == 3:
+                self.current_vis = VisList([i for i in self.columns], self)
             self._widget = rec_df.render_widget()
```

Comment:
is there a way we can write this such that we don't repeat the same code twice? You can make a quick function!

Reply:
yup!